### PR TITLE
React 18 & Patternfly 5 Upgrades

### PR DESCRIPTION
**Pull request by LightOfHeaven1994**
This PR contains:
    1. React router dom upgrade   v5 -> v6
    2. React upgrade                        17 -> 18
    3. Patternfly upgrade                 4 -> 5
    4. Other packages upgrade
    5. Couple minor backend bugfixes (this was needed to test Frontend changes)
    6. Small Frontend bugfixes/improvements
    7. Test fixes

### CODE REVIEW

**Comment by computercamplove** [comment]
@LightOfHeaven1994 For some reason alerts doesn't go away when i import archive
Screenshot from 2024-05-17 18-51-45, even if i navigate to different page, only when i reload page.
could you reproduce it?

**Comment by computercamplove** [comment]
2. Jenkins heatmap is not working for me, i copied same settings from production
image

**Comment by mshriver** [comment]
@LightOfHeaven1994 thanks for picking this up - Can you address @computercamplove concern above, and look at updating the frontend container to node:18 in this?

https://github.com/ibutsu/ibutsu-server/pull/498

I'm hitting _some_ of this in this MR, where I included a change to node:18 for the `ibutsu-pod.sh` script for local dev envs.

Node 16 is past end of life and we need to update it - I don't want 498 to completely turn into this PR, I started addressing some of the deprecations and then saw you're doing that here.

I'll keep node:16 in MR498's branch and then can rebase on top of this?

**Comment by LightOfHeaven1994** [comment]
> @LightOfHeaven1994 For some reason alerts doesn't go away when i import archive Screenshot from 2024-05-17 18-51-45, even if i navigate to different page, only when i reload page. could you reproduce it?

I can reproduce it. Will take a look :+1:

**Comment by LightOfHeaven1994** [comment]
> 2. Jenkins heatmap is not working for me, i copied same settings from production
>    image
> 
> **update**: the issue is not in widget but in what metadata i use - it works for singe plugin, but not for `common-pipelines/{prod-test-suite}`

This happens because you probably used "Project 1" when you were uploading results, but then you switched to another project that has no these uploads (no data). This is not a bug but it's just how ibutsu RBAC works

File: frontend/src/components/classification-dropdown.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ import {
   Dropdown,
   DropdownItem,
   DropdownToggle
-} from '@patternfly/react-core';
+} from '@patternfly/react-core/deprecated';
```
Comment (mshriver) [inline frontend/src/components/classification-dropdown.js:8]: Why are we pulling from deprecated here?

File: frontend/src/components/ibutsu-header.js [hunk 2]
```diff
@@ -187,55 +205,154 @@ export class IbutsuHeader extends React.Component {
...
+            toggle={this.toggle}
           >
-            {this.state.projects.map(project => (
-              <SelectOption key={project.id} value={projectToOption(project)} description={project.name} />
-            ))}
+            <SelectList id="select-typeahead-listbox">
+              {(projects.length === 0 && !filterValue) && (
+                <SelectOption
+                  isDisabled={true}
+                  description={"Ask Ibutsu admins to assing you a project"}>
```
Comment (mshriver) [inline frontend/src/components/ibutsu-header.js:293]: "Ask Ibutsu admins to add you to a project"

Comment (mshriver) [reply frontend/src/components/ibutsu-header.js:293]: Or maybe you were going for "Ask Ibutsu admins to assign you a project", that works too

File: frontend/src/components/classification-dropdown.js (hunk 1 above)
Comment (LightOfHeaven1994) [reply frontend/src/components/classification-dropdown.js]: @mshriver fixed

**Comment by LightOfHeaven1994** [comment]
> @LightOfHeaven1994 For some reason alerts doesn't go away when i import archive Screenshot from 2024-05-17 18-51-45, even if i navigate to different page, only when i reload page. could you reproduce it?

fixed :)

**Review by mshriver — APPROVED** [review]
There is obviously a ton here, but deployed and running locally with some test results imported and a dashboard created with a few widgets, everything is looking great!

The header is behaving well, and the improved rendering on the test result json object is a nice treat.

Thanks for maintaining Ibutsu and managing this upgrade - and all of the dependency resolution headaches that came with it!

I'm going to immediately start to take advantage of both the react-router upgrade's navigation hooks, and some of the PF5 components changes for the portal dashboards.

**Pull request merged**